xpart/rng_src/base_rng.h

```diff
@@ -28,9 +28,6 @@ double rng_get (uint32_t *s1, uint32_t *s2, uint32_t *s3, uint32_t *s4 ){
 
 /*gpufun*/
 void rng_set (uint32_t *s1, uint32_t *s2, uint32_t *s3, uint32_t *s4, uint32_t s ){
-  if (s == 0)
-    s = 1;      /* default seed is 1 */
-
```

Comment:
I believe this is done because zero is not a valid seed. We are basically saying that 0 and 1 are the same seed.
If you want to avoid this behavior, you need to assert on the python size that no seed is zero

Reply:
I assumed this was the reasoning, however, I don't see any reason why 0 would be invalid. Two lines below, if the LCG result is smaller than 2, 2 is added, making 0 a perfectly valid seed as well.

I tested it, and seed 0 creates perfectly valid results, uncorrelated from the other seeds.

Is this maybe a leftover from a less complete implementation of the rng?

Reply:
Ok, I agree. I will merge this